pkg/segment/query/querystatus.go

```diff
@@ -42,7 +44,10 @@ var numStates = 4
 
 const MAX_GRP_BUCKS = 3000
 const CANCEL_QUERY_AFTER_SECONDS = 5 * 60 // If 0, the query will never timeout
-const MAX_RUNNING_QUERIES = 32
+var MAX_RUNNING_QUERIES = runtime.GOMAXPROCS(0)
+
+const MAX_WAITING_QUERIES = 100
+const QUERY_MEM = 200 * 1024 * 1024 // 200MB
```

Comment:
make this mem configurable, so that operators can tune this for differently sized systems

Reply:
addressed in the latest commit.